ItemGroup info on getItemDetail

itisluiz opened the pull request:
Get information (id, displayName) pairs for all ItemGroups (creative tabs) an item pertains to. Very useful for categorizing items.

This is similar to what my recently closed (due to the api changes) pr had, but it now returns both the ID and display name of a given item group. Could go back to returning just the IDs if it feels too bloated. Also updates the documentation accordingly and adds an example of usage.

SquidDev commented on src/main/java/dan200/computercraft/shared/peripheral/generic/data/ItemData.java:
```diff
@@ -116,6 +118,28 @@ private static ITextComponent parseTextComponent( @Nonnull INBT x )
+        {
+            Map<String, Object> group = new HashMap<>( 2 );
+            group.put( "id", g.getId() );
+            group.put( "displayName", g.getDisplayName().getString() );
```
On 1.16, `getDisplayName` is client only. I think we'll need an access transformer here to access the underlying `displayName` field.

itisluiz replied:
No longer uses `getDisplayName` or `getId`, just the exposed `langId`

SquidDev commented on src/main/java/dan200/computercraft/shared/peripheral/generic/data/ItemData.java:
```diff
@@ -116,6 +118,28 @@ private static ITextComponent parseTextComponent( @Nonnull INBT x )
+        for ( ItemGroup g : stack.getItem().getCreativeTabs() )
+        {
+            Map<String, Object> group = new HashMap<>( 2 );
+            group.put( "id", g.getId() );
```
It feels like it's probably more useful to access expose the `langId` rather than the numerical ID - AFAIK the latter can change between runs.

itisluiz replied:
Done, now exposes `langId` through an access transformer and returns a table of them, in a similar way to item `tags`.

SquidDev commented:
Thanks! Just left a couple of comments, but otherwise looks good.

itisluiz committed:
ItemData now contains langId

Instead of returning the ItemGroup ID and displaName, getItemDetail() now returns a list of unlocalized item group names, using an AT for getting the langId from ItemGroup.

itisluiz commented:
Thanks for the feedback! Sorry for making the pull request so all over the place for such a simple change.

itisluiz committed:
Fix missing line break

Adds missing line break at accesstransformer.cfg

SquidDev committed:
Expose both item group and display name

SquidDev reviewed (approved):
Thanks for the quick change, sorry for being so slow on my end!

SquidDev committed:
Fix code example post me interferring
